go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/cilium/hive
 
-go 1.21.3
+go 1.24.1
+
+toolchain go1.24.4
```

Comment:
why mixed 1.24.1 and 1.24.4?

Reply:
:thinking: Not sure what happened, the tooling this this modification. I cannot even remember why I needed the newer go version anymore.

Reply:
Ah, so I wanted to use `maps.Keys` which got added to the stdlib in 1.23, not sure why i picked `1.24.1`. When you use the `go` command to upgrade, it will place the go version you have installed in the `toolchain`. As I understand it the `go` command informs the stdlib version and the `toolchain` the compiler version ect. I think we can leave the toolchain out as we do not have specific toolchain requirements that I am aware of.

Also, since we only need >=1.23, I will change it to that.